src/index.ts

```diff
@@ -100,6 +82,29 @@ export default class Kumulos {
         }
     }
 
+    private initializePushFeature() {
+        trackOpenFromQuery(this.context);
+        registerServiceWorker(this.context.serviceWorkerPath);
+
+        new PromptManager(
+            this,
+            this.context,
+            this.rootFrame
+        );
+
+        this.maybeAddMessageEventListenerToSW();
+        this.maybeFireOpenedHandler();
+    }
+
+    private maybeAddMessageEventListenerToSW() {
+        if ('serviceWorker' in navigator) {
+            navigator.serviceWorker.addEventListener(
```

Comment:
Now this is refactored, maybe we can invert condition & early return to reduce nesting of the block?

Reply:
Done